src/filter/GaussianFilter.js

```diff
@@ -0,0 +1,58 @@
+import IJ from '../ij';
+
+export default function gaussian_filter(n, sigma){
+	var cols = this.width;
+	var new_img = this.clone();
+	var kernel = get_gaussian_kernel(n, sigma);
+	var temp = cols*n; //variable for checking the image border
+	for(var p = cols*n+n; p < this.data.length - cols*n; p++){
+		if(p == temp+cols-n){
+			p += 2*n - 1;
+		}else{
+			new_img.data[p] = convolution(p, this.data, kernel);
+		}
+	}
+}
+
+function get_gaussian_kernel(n, sigma){
+	var kernel = [];
+	var sigma2 = 2*(sigma*sigma);//2*sigma^2
+	var sum = 0;
+	for(var y = -n; y <= n; y++){
+		for(var x = -n; x <= n; x++){
+			var value = -((x*x) + (y*y))/sigma2;
+			value = Math.pow(Math.E, value);
+			sum += value;
+			kernel.push(value);
+		}
+	}
+
+	for(var i = 0; i < kernel.length; i++){
+		kernel[i] /= sum;
+	}
+	return kernel;
+}
+
+function convolution(p, data, kernel){
```

Comment:
You should first create a convolution filter. This was referenced in #33. 
It would be nice if the algorithm would allow not only square matrix but in reality and kind of 2D array !!!
You could take the opportunity to use the getDataXY and setDataXY that may be easier to understand the code.